internal/agent/runtime/internal/failure_files.go

```diff
@@ -0,0 +1,104 @@
+package internal
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/tinkerbell/tink/internal/agent/failure"
+)
+
+// NewFailureFiles creates a new FailureFiles instance with isolated underlying files. Consumers
+// are responsible for calling FailureFiles.Close().
+func NewFailureFiles() (*FailureFiles, error) {
+	reason, err := os.CreateTemp("", "failure-reason-*")
+	if err != nil {
+		return nil, err
+	}
+
+	message, err := os.CreateTemp("", "failure-message-*")
+	if err != nil {
+		return nil, err
+	}
+
+	return &FailureFiles{
+		reason:  reason,
+		message: message,
+	}, nil
+}
+
+// FailureFiles provides mountable files for runtimes that can be used to extract
+// a reason and message from actions.
+type FailureFiles struct {
+	reason  *os.File
+	message *os.File
+}
+
+// Close closes all files tracked by f.
+func (f *FailureFiles) Close() error {
+	os.Remove(f.reason.Name())
```

Comment:
if the model is going to be to not capture the errors here, maybe we should change the interface to not return an error?

Reply:
Also, the name `Close` actually doesn't feel accurate here. deleting files is different than closing them. is this satisfying an interface method?